pkg/structs/builder.go

```diff
@@ -103,7 +112,17 @@ func UpsertBuilderByFields(name string, fields []Field) (*Builder, error) {
 		if field.Validator != nil {
 			builder.validators[field.Name] = field.Validator
 		}
+
 		builder.fieldIndexMap[field.Name] = i
+
+		if field.Anonymous && typ.Kind() == reflect.Struct {
+			for j := 0; j < typ.NumField(); j++ {
+				if typ.Field(i).Anonymous {
```

Comment:
```suggestion
				if typ.Field(j).Anonymous {
```
对于匿名结构体的场景可以考虑把value改成数组记录一下它们的外层和内层的index，然后多层嵌套的场景现在是直接跳过了，是不是也应该记录一下

Reply:
已修改为数组